### ISSUE TITLE
Client connection timeout is not working

### ISSUE THREAD

**Issue by xavierhardy** [davebshow/aiogremlin#4]
Try without any running Gremlin server the following commands:

```
loop = asyncio.get_event_loop()
ws = GremlinClient(url="ws://localhost:8182", loop=loop, timeout=100000)
loop.run_until_complete(ws.execute("1"))
```

It immediately raises an error. Moreover, it is not clear what unit timeout is in (I guess seconds, but I haven't seen any time units mentioned).

```
aiohttp.errors.ClientOSError: [Errno 111] Can not connect to localhost:8182 [Connect call failed ('127.0.0.1', 8182)]
```

When I try using a server that has run before and stopped running at some point, I get the same behavior.

**Comment by xavierhardy** [comment]
conn_timeout in the connector should be configurable. I also noticed that the problem actually came from the aiowebsocketclient library. Setting conn_timeout does not help but it should.

**Comment by davebshow** [comment]
Yes, `conn_timeout` should be configurable using the main `GremlinClient` interface (without configuring a `GremlinConnector` instance and passing directly as an arg to `GremlinClient`), I will fix this in the 0.1.3 release. 

I believe the connection timeout issue is occurring because `conn_timeout` is not being passed to the underlying `aiohttp.TCPConnector`. I have fixed this in the `aiowebsocketclient` 0.0.5 release. `aiogremlin` 0.1.3 reflects all of these changes. 

I don't have time today to implement tests for this behaviour, but please upgrade and try to see if this fixes your problem.

**Comment by xavierhardy** [comment]
Thanks, let me try again.

**Comment by xavierhardy** [comment]
I have updated `aiogremlin` and `aiowebsocketclient` and, yet, it does not work. The exception is still sent as soon as the connection fails, not waiting for `conn_timeout` seconds...

**Comment by davebshow** [comment]
Ok. I will look into this more thoroughly over the day or so.

**Comment by davebshow** [comment]
Hi there @xavierhardy. Sorry about the slow response. I looked a bit more into this issue, and it seems that this is the expected behaviour coming from `aiohttp` and `asycnio` when there is no server running. `aiohttp` uses the `asyncio.BaseEventLoop.create_connection` method to establish a connection, and when the server is not up this method immediately raises a `OSError`. `aiohttp` passes this error along as `ClientOSError`. 

Now, I don't mean to suggest that is what should happen, only that this implementation detail would need to be handled in the `aiohttp` source code...
